### PR TITLE
prepare webgme to start as a service with "systemd" the new "init"

### PR DESCRIPTION
**Pull request by phreed**
http://0pointer.de/blog/projects/socket-activation.html
https://www.freedesktop.org/software/systemd/man/systemd.socket.html
https://developer.atlassian.com/blog/2015/03/docker-systemd-socket-activation/

This makes use of https://www.npmjs.com/package/systemd-socket to adopt the systemd socket.
https://github.com/herzi/systemd-socket

The attached zip file contains sample systemd config files.

[init_systemd.zip](https://github.com/webgme/webgme/files/224751/init_systemd.zip)

There is another package which does similar things but uses monkey-patching.
https://github.com/rubenv/node-systemd

### CODE REVIEW

File: src/server/standalone.js [hunk 1]
```diff
@@ -194,9 +195,20 @@ function StandAloneServer(gmeConfig) {
...
+        // should be considered to shut the service down if it
+        // is not getting much use.
+
+        // https://github.com/herzi/systemd-socket
+        // https://nodejs.org/api/http.html#http_server_listen_handle_callback
+        __httpServer.listen(systemdSocket(0) || gmeConfig.server.port, function () {
         __httpServer.listen(gmeConfig.server.port, function () {
             // Note: the listening function does not return with an error, errors are handled by the error event
-            logger.debug('Http server is listening on ', {metadata: {port: gmeConfig.server.port}});
+            logger.error('Http server is listening on ',
```
Comment (pmeijer) [inline src/server/standalone.js]: It's loggin an error here..

**Comment by pmeijer** [comment]
The tests are failing with syntax errors in standalone. Btw you can make more commits to your branch with worrying about the history in the master branch in webgme; Github provides a "squash" option before merging (this came out last week or so).

File: src/server/standalone.js [hunk 2]
```diff
@@ -194,9 +195,19 @@ function StandAloneServer(gmeConfig) {
...
 
-        __httpServer.listen(gmeConfig.server.port, function () {
+        // something link
+        // https://github.com/rubenv/node-autoquit
+        // should be considered to shut the service down if it
+        // is not getting much use.
+
+        // https://github.com/herzi/systemd-socket
+        // https://nodejs.org/api/http.html#http_server_listen_handle_callback
+        __httpServer.listen(systemdSocket(0) || gmeConfig.server.port, function () {
```
Comment (brollb) [inline src/server/standalone.js]: It looks like `systemdSocket(0)` returns the port number... If so, why don't you just add this to your config files and set `gmeConfig.server.port` to this value?

Comment (phreed) [inline src/server/standalone.js]: Are you suggesting moving the systemd code to one of the 'config/config.*.js' files?
Yes, that would probably work fine.

Comment (brollb) [inline src/server/standalone.js]: yeah, that's what I mean. I don't like the idea of introducing a systemd specific dependency...

This should be doable similar to how you can allow users to set the port w/ env variables: https://github.com/dfst/deepforge/blob/master/config/config.base.js#L10

Comment (phreed) [inline src/server/standalone.js]: Sorry I took so long to get back to this.
The problem with the approach you advocate is that systemSocket(0) does not return a number.
systemSocket(0) returns an object of the form '{fd: 3}'.
This should be simple enough to fix.
The problem is in the validator, it does not like an argument of the form '{fd: 3}'.
I hacked webgme/config/validator.js:194
// assertNumber('config.server.port', config.server.port);
This allows the file descriptor adoption.

**Comment by pmeijer** [comment]
This is resolved in #1206

**Pull request closed without merging**